### PR TITLE
:tada: add multiplicative null

### PR DESCRIPTION
**Pull request by MichielStock**
# Adding a new null model

This samples from a joint distribution proportional with the project of the degree distributions of the two levels. This means that the two levels act independently: just proportional to the marginal activity of the species.

## Checklist

- [x] The code is covered by unit tests
  - [x] Additional cases or module in `test/...`
  - [x] Relevant lines in `test/runtests.jl`
- [x] The code is *documented*
  - [x] Docstrings written for every function
  - [x] If needed, additional lines in `docs/src/...`
- [x] All contributors have added their name and affiliation to `.zenodo.json`

## Pinging

Pinging @tpoisot

### CODE REVIEW

File: src/rand/null.jl [hunk 1]
```diff
@@ -43,3 +43,22 @@ species.
...
+
+
+"""
+    null2mult(N::BinaryNetwork)
+
+Given a matrix `A`, `null2mult(A)` returns a matrix with the same dimensions, where
+every interaction happens with a probability equal to the product of the degree
+of each species.
+"""
+function null2mult(N::BinaryNetwork)
```
Comment (tpoisot) [inline src/rand/null.jl]: I would give it a new name -- it's really different from `null2`, which is doing averages. What about `null4`? At this point, we might as well call this Type IV null model.

File: src/rand/null.jl [hunk 2]
```diff
@@ -43,3 +43,22 @@ species.
...
+  A = N.A
+  n, m = size(A)
+  Afiltered = sum(A, dims=1) .* sum(A, dims=2) ./ sum(A)^2
+  if typeof(N) <: AbstractBipartiteNetwork
+  return BipartiteProbabilisticNetwork(Afiltered)
```
Comment (tpoisot) [inline src/rand/null.jl]: We would like (ideally) to keep the same species, so this line should be

~~~
BipartiteProbabilisticNetwork(Afiltered, species_objects(N)...)
~~~

File: src/rand/null.jl [hunk 3]
```diff
@@ -43,3 +43,22 @@ species.
...
+  else
+    return UnipartiteProbabilisticNetwork(Afiltered)
```
Comment (tpoisot) [inline src/rand/null.jl]: Same comment as above. One way to solve both would be to do 

~~~
ReturnType = typeof(N) <: AbstractBipartiteNetwork ? BipartiteProbabilisticNetwork : UnipartiteProbabilisticNetwork
return ReturnType(Afiltered, species_objects(N)...)
~~~

File: src/rand/null.jl [hunk 4]
```diff
@@ -43,3 +43,22 @@ species.
...
+"""
+    null2mult(N::BinaryNetwork)
+
+Given a matrix `A`, `null2mult(A)` returns a matrix with the same dimensions, where
+every interaction happens with a probability equal to the product of the degree
+of each species.
+"""
+function null2mult(N::BinaryNetwork)
+  A = N.A
+  n, m = size(A)
```
Comment (tpoisot) [inline src/rand/null.jl]: these aren't used

File: src/rand/null.jl [hunk 5]
```diff
@@ -43,3 +43,22 @@ species.
...
+  Afiltered = sum(A, dims=1) .* sum(A, dims=2) ./ sum(A)^2
```
Comment (tpoisot) [inline src/rand/null.jl]: you can do `sum(N)` directly I think, and if not it's a matter of changing the method for `sum` to allow a `dim` keyword like `degree` does.

File: src/rand/null.jl (hunk 1 above)
Comment (MichielStock) [reply src/rand/null.jl]: done

File: src/rand/null.jl (hunk 5 above)
Comment (MichielStock) [reply src/rand/null.jl]: is done and tested

File: src/links/connectance.jl [hunk 6]
```diff
@@ -1,15 +1,20 @@
...
-    sum(N::AbstractEcologicalNetwork)
+    sum(N::AbstractEcologicalNetwork; dims=nothing)
 
 This function will return the sum of all interactions in the network. For
 quantitative networks, this is the sum of interaction strengths. For binary
 networks, this is the number of interactions. For probabilistic networks, this
 is the expected number of realized interactions.
+
+Optionally, one can give the argument dims, simular to the native `sum`, which
+computes the sum of the interactions for the lower (`sum=2`) or higher (`sum=1`)
```
Comment (tpoisot) [inline src/links/connectance.jl]: should be `dims` and not `sum`

File: src/links/connectance.jl [hunk 7]
```diff
@@ -1,15 +1,20 @@
...
+trophic level.
 """
-function sum(N::AbstractEcologicalNetwork)
-   return sum(N.A)
+function sum(N::AbstractEcologicalNetwork; dims::Union{Nothing,Int}=nothing)
+   (dims == nothing && return sum(N.A)) || return sum(N.A, dims=dims)
```
Comment (tpoisot) [inline src/links/connectance.jl]: Do you think it would be wise to add an `@assert` to check that `dims` is only 1 or 2?

**Comment by tpoisot** [comment]
You'll also need to rebase `develop` .

File: src/links/connectance.jl (hunk 7 above)
Comment (MichielStock) [reply src/links/connectance.jl]: Will do, though default behaviour is to just return the array itself if dims > ndims.

Comment (tpoisot) [reply src/links/connectance.jl]: Agreed, but most users will not know that.

**Comment by tpoisot** [comment]
OK, rebase and we're :+1: to merge

**Comment by MichielStock** [comment]
I think the rebase is done now.

File: Project.toml [hunk 8]
```diff
@@ -8,6 +8,7 @@ Combinatorics = "861a8166-3701-5b0c-9a16-15d98fcdc6aa"
 DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"
 DelimitedFiles = "8bb1440f-4735-579b-a4ab-409b98df4dab"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
+Luxor = "ae8d54c2-7ccd-5906-9d76-62fc9837b5bc"
```
Comment (tpoisot) [inline Project.toml]: This shouldn't be needed

**Comment by tpoisot** [comment]
These three files have conflicts
~~~
Manifest.toml
docs/src/properties/randomization.md
src/links/connectance.jl 
~~~

**Comment by tpoisot** [comment]
I've fixed the final merge conflicts -- thanks!

**Pull request merged**